great_expectations/datasource/fluent/interfaces.py

```diff
@@ -693,6 +694,7 @@ def _execution_engine_type(self) -> Type[_ExecutionEngineT]:
         """Returns the execution engine to be used"""
         return self.execution_engine_override or self.execution_engine_type
 
+    @public_api
```

Comment:
I don't think this one should be in the public API. The method of the same name on Asset should be though!